Source/ASNetworkImageNode.mm

```diff
@@ -149,20 +154,28 @@ - (void)setURL:(NSURL *)URL resetToDefault:(BOOL)reset
       return;
     }
     
-    [self _cancelImageDownload];
+    {
+      ASDN::MutexUnlocker l(__instanceLock__);
+      [self _cancelImageDownload];
+    }
+    
     _imageLoaded = NO;
     
     _URL = URL;
     
-    BOOL hasURL = _URL == nil;
+    BOOL hasURL = (_URL == nil);
     if (reset || hasURL) {
-      [self _setImage:_defaultImage];
-      /* We want to maintain the order that currentImageQuality is set regardless of the calling thread,
-       so always use a dispatch_async to ensure that we queue the operations in the correct order.
-       (see comment in displayDidFinish) */
-      dispatch_async(dispatch_get_main_queue(), ^{
-        self.currentImageQuality = hasURL ? 0.0 : 1.0;
-      });
+      UIImage *defaultImage = _defaultImage;
+      {
+        ASDN::MutexUnlocker l(__instanceLock__);
+        [self _setImage:defaultImage];
```

Comment:
Should we have another version of this method? `_locked_setImage:` ? I think we may want to get the default image and set it in one locked transaction.